### ISSUE TITLE
Race condition about embed_genesis.cmake

### ISSUE THREAD

**Issue by abitmore** [bitshares/bitshares-core#2188]
### Build Error Description

When building with multiple cores, there is probably a race condition that multiple `embed_genesis.cmake` instances may be started at the same time and try to write to the same destination files, which leads to `configure_file Problem configuring file` errors.

### Build Environment
Github Autobuild.

### Console Logs (optional)
Ubuntu release build of https://github.com/bitshares/bitshares-core/runs/724112359
```
[ 65%] Linking CXX static library libgraphene_chain.a
make[2]: Leaving directory '/home/runner/work/bitshares-core/bitshares-core/_build'
[ 65%] Built target graphene_chain
make[2]: Entering directory '/home/runner/work/bitshares-core/bitshares-core/_build'
make[2]: Entering directory '/home/runner/work/bitshares-core/bitshares-core/_build'
[ 65%] Generating egenesis_brief.cpp, egenesis_full.cpp
Scanning dependencies of target graphene_egenesis_none
make[2]: Leaving directory '/home/runner/work/bitshares-core/bitshares-core/_build'
make[2]: Entering directory '/home/runner/work/bitshares-core/bitshares-core/_build'
[ 65%] Building CXX object libraries/egenesis/CMakeFiles/graphene_egenesis_none.dir/egenesis_none.o
[ 65%] Linking CXX static library libgraphene_egenesis_none.a
make[2]: Leaving directory '/home/runner/work/bitshares-core/bitshares-core/_build'
[ 65%] Built target graphene_egenesis_none
make[2]: Entering directory '/home/runner/work/bitshares-core/bitshares-core/_build'
[ 65%] Generating egenesis_brief.cpp, egenesis_full.cpp
-- Generating egenesis
-- Chain-id: 4018d7844c78f6a6c41c6a552b898022310fc5dec06da467ee7905a8dad512c8
-- Generating egenesis
-- Chain-id: 4018d7844c78f6a6c41c6a552b898022310fc5dec06da467ee7905a8dad512c8
CMake Error at /home/runner/work/bitshares-core/bitshares-core/libraries/egenesis/embed_genesis.cmake:11 (configure_file):
  configure_file Problem configuring file


make[2]: *** [libraries/egenesis/egenesis_brief.cpp] Error 1
make[2]: *** Deleting file 'libraries/egenesis/egenesis_brief.cpp'
libraries/egenesis/CMakeFiles/graphene_egenesis_brief.dir/build.make:80: recipe for target 'libraries/egenesis/egenesis_brief.cpp' failed
make[1]: *** [libraries/egenesis/CMakeFiles/graphene_egenesis_brief.dir/all] Error 2
make[2]: Leaving directory '/home/runner/work/bitshares-core/bitshares-core/_build'
make[1]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:1266: recipe for target 'libraries/egenesis/CMakeFiles/graphene_egenesis_brief.dir/all' failed
Scanning dependencies of target graphene_egenesis_full
make[2]: Leaving directory '/home/runner/work/bitshares-core/bitshares-core/_build'
make[2]: Entering directory '/home/runner/work/bitshares-core/bitshares-core/_build'
[ 65%] Generating egenesis_brief.cpp, egenesis_full.cpp
-- Generating egenesis
-- Chain-id: 4018d7844c78f6a6c41c6a552b898022310fc5dec06da467ee7905a8dad512c8
[ 66%] Building CXX object libraries/egenesis/CMakeFiles/graphene_egenesis_full.dir/egenesis_full.o
[ 66%] Linking CXX static library libgraphene_egenesis_full.a
make[2]: Leaving directory '/home/runner/work/bitshares-core/bitshares-core/_build'
[ 66%] Built target graphene_egenesis_full
make[1]: Leaving directory '/home/runner/work/bitshares-core/bitshares-core/_build'
make: *** [all] Error 2
Makefile:146: recipe for target 'all' failed
make: Leaving directory '/home/runner/work/bitshares-core/bitshares-core/_build'
##[error]Process completed with exit code 2.
```

## CORE TEAM TASK LIST
- [ ] Evaluate `Build Error`
- [ ] Provide build guidance
- [ ] <OR> Create `Bug Report`

**Comment by abitmore** [comment]
It's written in the [`cmake` documentation](https://cmake.org/cmake/help/v3.13/command/add_custom_command.html):
> add_custom_command ... is for adding a custom command to produce an output ... Do not list the output in more than one independent target that may build in parallel or the two instances of the rule may conflict (instead use the add_custom_target() command to drive the command and make the other targets depend on that one).

While we have code: https://github.com/bitshares/bitshares-core/blob/318aba2c066b1d8c15be00b1c5d46483521ce0aa/libraries/egenesis/CMakeLists.txt#L9-L12 https://github.com/bitshares/bitshares-core/blob/318aba2c066b1d8c15be00b1c5d46483521ce0aa/libraries/egenesis/CMakeLists.txt#L27-L30

**Comment by abitmore** [comment]
Fixed by #2250.
